### ISSUE TITLE
fail on zero tests found

### ISSUE THREAD

**Issue by pytestbot** [pytest-dev/pytest#662]
Originally reported by: **Boris Rybalkin (BitBucket: [cyberb](http://bitbucket.org/cyberb), GitHub: [cyberb](http://github.com/cyberb))**

----------------------------------------

Sometimes it is critical to fail when py.test does not find any test. This may happen when you are accidentally running in the wrong directory on the build agent and may lead to the impression that everything is ok.

Something like:
py.test --fail-on-empy

EDIT: fail by default would be even better

----------------------------------------
- Bitbucket: https://bitbucket.org/pytest-dev/pytest/issue/662

**Comment by The-Compiler** [comment]
There's a pull request in #817 now.

**Comment by vladyslav-savchenko** [comment]
Hi People,

I think it seems to be a bug here: _pytest/main.py:614..620:
          if self._notfound:                                              
              errors = []                                                 
              for arg, exc in self._notfound:                             
                  line = "(no name %r in any of %r)" % (arg, exc.args[0]) 
                  errors.append("not found: %s\n%s" % (arg, line))        
                  #XXX: test this                                         
              raise pytest.UsageError(*errors)
If no tests found, "python -m pytest <py_src>"  returns exit_code 4(EXIT_USAGEERROR) instead of 5 (EXIT_NOTESTSCOLLECTED).

**Comment by RonnyPfannschmidt** [comment]
@vladyslav-savchenko that is intentional - passing missing files is a usage error, passing test filters that cause no tests to be is the notestscollected error

**Comment by vladyslav-savchenko** [comment]
No, files exist, I have checked that with os.path.exists.
But here (_pytest/main.py:714) "matching" and "names" are empty and "NoMatch" exception raises.

I have found the cause of this wrong behavior here (_pytest/python.py:154). I get an exit code 4 if a python source does not have an extension at all. I think it is not completely correct to detect if it is a python source just checking for its extension. Not every python script has such one, but command
`> file <path>`
returns nevertheless
`<path>: Python script, ASCII text executable`.

Unfortunately I can not suggest a better solution yet :- (

Thank you!
